Storagekey on primary key

I tried using the StorageKey method on primary key as follows

```
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Uint64("id").StorageKey("user_id"),
	}
}

func (User) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("pets", Pet.Type),
	}
}

func (Pet) Fields() []ent.Field {
	return []ent.Field{
		field.Uint64("id").StorageKey("pet_id"),
	}
}

func (Pet) Edges() []ent.Edge {
	return []ent.Edge{
		edge.From("owner", User.Type).
			Ref("pets").
			Required().
			Unique(),
	}
}
```

when I tried to query the pet's owner as follows

```
user := client.User.Create().SaveX(ctx)
pet := client.Pet.Create().SetOwner(user).SaveX(ctx)

pet.QueryOwner().OnlyX(ctx)
```
I got an error

```
2020/07/18 20:14:55 driver.Query: query=SELECT DISTINCT `users`.`user_id` FROM `users` JOIN (SELECT `user_pets` FROM `pets` WHERE `pet_id` = ?) AS `t1` ON `users`.`pet_id` = `t1`.`user_pets` LIMIT ? args=[3 2]
panic: Error 1054: Unknown column 'users.pet_id' in 'on clause'
```

--- a/dialect/sql/sqlgraph/graph.go
+++ b/dialect/sql/sqlgraph/graph.go
@@ -164,7 +164,7 @@ func Neighbors(dialect string, s *Step) (q *sql.Selector) {
 		q = builder.Select().
 			From(t1).
 			Join(t2).
-			On(t1.C(s.From.Column), t2.C(s.Edge.Columns[0]))
+			On(t1.C(s.To.Column), t2.C(s.Edge.Columns[0]))
 	case r == O2M || (r == O2O && !s.Edge.Inverse):
 		q = builder.Select().
 			From(builder.Table(s.To.Table)).

--- a/entc/integration/customid/ent/migrate/schema.go
+++ b/entc/integration/customid/ent/migrate/schema.go
@@ -97,7 +97,7 @@ var (
 	}
 	// UsersColumns holds the columns for the "users" table.
 	UsersColumns = []*schema.Column{
-		{Name: "id", Type: field.TypeInt, Increment: true},
+		{Name: "oid", Type: field.TypeInt, Increment: true},
 		{Name: "user_children", Type: field.TypeInt, Nullable: true},
 	}
 	// UsersTable holds the schema information for the "users" table.

--- a/entc/integration/customid/ent/schema/user.go
+++ b/entc/integration/customid/ent/schema/user.go
@@ -18,7 +18,8 @@ type User struct {
 // Fields of the User.
 func (User) Fields() []ent.Field {
 	return []ent.Field{
-		field.Int("id"),
+		field.Int("id").
+			StorageKey("oid"),
 	}
 }
 

--- a/entc/integration/customid/ent/user/user.go
+++ b/entc/integration/customid/ent/user/user.go
@@ -10,7 +10,7 @@ const (
 	// Label holds the string label denoting the user type in the database.
 	Label = "user"
 	// FieldID holds the string denoting the id field in the database.
-	FieldID = "id"
+	FieldID = "oid"
 
 	// EdgeGroups holds the string denoting the groups edge name in mutations.
 	EdgeGroups = "groups"

